### PR TITLE
add new component of login options

### PR DESCRIPTION
**Pull request by OshriAsulin**
I added a component to the navbar that lets more options in the future user

When the button is clicked the user's name will be displayed, with additional options when the user clicks the button

### CODE REVIEW

**Comment by idodav** [comment]
@OshriAsulin please notice that you have conflicts.
Please fix them.
Also I'd love it if you could add a screenshot :)

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+
+import React, { useState } from "react";
+import { AppBar, Avatar, Button, Menu, MenuItem, Tooltip, Typography } from "@mui/material";
+import IconButton from '@mui/material/IconButton';
+
+import Box from '@mui/material/Box';
+import { Login } from "@mui/icons-material";
+
+type Props = {
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:9]: Remove the interface if it's empty pelase

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+
+import React, { useState } from "react";
+import { AppBar, Avatar, Button, Menu, MenuItem, Tooltip, Typography } from "@mui/material";
+import IconButton from '@mui/material/IconButton';
+
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:5]:
```suggestion
```

Remove the new line please

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:1]:
```suggestion
```
Remove the new line pelase

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import Box from '@mui/material/Box';
+import { Login } from "@mui/icons-material";
+
+type Props = {
+    // Define your component's props here
+};
+
+const LoginComponent: React.FC<Props> = (props) => {
+    //this login state is example for auth user
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:14]: remove unnecessary comments please

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                            <span style={{ color: '#90caf9' }}>User </span>
+                            <Tooltip title="Open settings">
+                                <IconButton sx={{ p: 0 }}>
+                                    <Avatar alt="Remy Sharp" sx={{ width: 34, height: 34 }} />
+                                </IconButton>
+                            </Tooltip>
+                            <Menu
+                                sx={{ mt: '55px' }}
+                                id="menu-appbar"
+                                // anchorEl={anchorElUser}
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:43]:
```suggestion
```

Remove unused line please

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        setAnchorElUser(!anchorElUser);
+    };
+
+    return (
+        <div>
+            {
+                login ?
+                    <Button variant="outlined" onClick={() => setLogin(!login)}>Login</Button> :
+                    (<div>
+                        <Box onClick={handleOpenUserMenu} sx={{ flexGrow: 0, border: 1, borderColor: 'primary.main', cursor: 'pointer', padding: "5px", borderRadius: "10px" }}>
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:33]: I'd love it if you could write the CSS in a separate css.module file.

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:1]: If you don't use already 
Please install and use prettier
https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

**Review by idodav — CHANGES_REQUESTED** [review]

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                                anchorOrigin={{
+                                    vertical: 'top',
+                                    horizontal: 'right',
+                                }}
+                                keepMounted
+                                transformOrigin={{
+                                    vertical: 'top',
+                                    horizontal: 'right',
+                                }}
+                                open={Boolean(anchorElUser)}
```
Comment (nkotzrenko) [inline traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:48]: Any reason to use the Boolean constructor?

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 1 above)
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:9]: Ok i will fix this

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:1]: I was install this now

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 4 above)
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:9]: this comment is to explain this state that will be deleted in the future

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 7 above)
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:48]: The open attribute is used to control whether the menu is open or closed. It expects a Boolean value (true or false) to determine the state of the menu.
 So, the open attribute will be set to true when anchorElUser has a value and false when anchorElUser is false.
 
And if I remove this Boolean in the open attribute is makes an error

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 5 above)
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:38]: You right I remove it for clean the code

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 6 above)
Comment (OshriAsulin) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:28]: Ok, i will do that

**Comment by OshriAsulin** [comment]
I want to ask about the convention for the name of the component, What I do in Camel case ok?

or do you want it will be in the kebab case like all the other components?

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 7 above)
Comment (nkotzrenko) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx:48]: What error does it raise? As far as I understand the property accepts type boolean and you pass it a boolean anyway.

**Comment by idodav** [comment]
> I want to ask about the convention for the name of the component, What I do in Camel case ok?
> 
> or do you want it will be in the kebab case like all the other components?

File names should be kebab-case.
Component names should be PascalCase.

File: traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx (hunk 4 above)
Comment (idodav) [reply traffic-dashboard/src/components/LoginComponent/LoginComponent.tsx]: okay

File: traffic-dashboard/src/components/LoginComponent/login-component.module.scss [hunk 8]
```diff
@@ -0,0 +1,15 @@
+$primary-color: #90caf9;
+
+.my_box {
+  flex-grow: 0;
+  border: 1px solid $primary-color; // Assuming "primary.main" is a valid variable
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/login-component.module.scss:5]: remove comment please

File: traffic-dashboard/src/components/LoginComponent/login-component.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        </div>
+      )}
+    </div>
+  );
+};
+
+export default LoginComponent;
+
+
+// sx={{
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/login-component.tsx:73]: remove commented code

File: traffic-dashboard/src/components/LoginComponent/login-component.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  Menu,
+  MenuItem,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import IconButton from "@mui/material/IconButton";
+import styles from './login-component.module.scss'
+
+const LoginComponent: React.FC<{}> = (props) => {
+  //this login state is example for auth user
```
Comment (idodav) [inline traffic-dashboard/src/components/LoginComponent/login-component.tsx:14]: remove comment please

**Review by idodav — CHANGES_REQUESTED** [review]

**Review by idodav — APPROVED** [review]

**Review by idodav — APPROVED** [review]

**Pull request merged**